qiskit/algorithms/minimum_eigensolvers/vqe.py

```diff
@@ -0,0 +1,333 @@
+# that they have been altered from the originals.
+
+"""The variational quantum eigensolver algorithm."""
+
+from __future__ import annotations
+
+import logging
+from time import time
+from collections.abc import Callable, Sequence
+
+import numpy as np
+
+from qiskit.algorithms.gradients import BaseEstimatorGradient
+from qiskit.circuit import QuantumCircuit
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from ..exceptions import AlgorithmError
+from ..list_or_dict import ListOrDict
+from ..optimizers import Optimizer, Minimizer, OptimizerResult
+from ..variational_algorithm import VariationalAlgorithm, VariationalResult
+from .minimum_eigensolver import MinimumEigensolver, MinimumEigensolverResult
+from ..observables_evaluator import estimate_observables
+from ..utils import validate_initial_point, validate_bounds
+
+logger = logging.getLogger(__name__)
+
+
+class VQE(VariationalAlgorithm, MinimumEigensolver):
+    r"""The variational quantum eigensolver (VQE) algorithm.
+
+    VQE is a hybrid quantum-classical algorithm that uses a variational technique to find the
+    minimum eigenvalue of a given Hamiltonian operator :math:`H`.
+
+    The VQE algorithm is executed using an :attr:`estimator` primitive, which must be specified as
+    the first argument on instantiation.
+
+    An instance of ``VQE`` also requires an :attr:`ansatz` parameterized :class:`.QuantumCircuit` to
+    prepare the trial state :math:`|\psi(\vec\theta)\rangle`, as well as a classical
+    :attr:`optimizer`. The optimizer varies the circuit parameters :math:`\vec\theta` such that the
+    expectation value of the operator on the corresponding state approaches a minimum,
+
+    .. math::
+
+        \min_{\vec\theta} \langle\psi(\vec\theta)|H|\psi(\vec\theta)\rangle.
+
+    The Estimator is used to compute this expectation value for every optimization step.
+
+    The optimizer can either be one of Qiskit's optimizers, such as
+    :class:`~qiskit.algorithms.optimizers.SPSA` or a callable with the following signature:
+
+    .. code-block:: python
+
+        from qiskit.algorithms.optimizers import OptimizerResult
+
+        def my_minimizer(fun, x0, jac=None, bounds=None) -> OptimizerResult:
+            # Note that the callable *must* have these argument names!
+            # Args:
+            #     fun (callable): the function to minimize
+            #     x0 (np.ndarray): the initial point for the optimization
+            #     jac (callable, optional): the gradient of the objective function
+            #     bounds (list, optional): a list of tuples specifying the parameter bounds
+
+            result = OptimizerResult()
+            result.x = # optimal parameters
+            result.fun = # optimal function value
+            return result
+
+    The above signature also allows one to directly pass any SciPy minimizer, for instance as
```

Comment:
```suggestion
    The above signature also allows one to use any SciPy minimizer, for instance as
```
When it involves wrapping it as a partial `directly pass` does seem a little odd